src/drivers/intel/dmic.c

```diff
@@ -134,9 +134,8 @@ static const uint32_t coef_base_a[4] = {PDM0_COEFFICIENT_A, PDM1_COEFFICIENT_A,
 static const uint32_t coef_base_b[4] = {PDM0_COEFFICIENT_B, PDM1_COEFFICIENT_B,
 	PDM2_COEFFICIENT_B, PDM3_COEFFICIENT_B};
 
-/* Global configuration request for DMIC, need to use uncached address to access them */
-static SHARED_DATA struct sof_ipc_dai_dmic_params *dmic_prm[DMIC_HW_FIFOS];
-static SHARED_DATA int dmic_active_fifos_mask;
+/* Global configuration request and state for DMIC */
+static SHARED_DATA struct dmic_global_shared dmic_global;
```

Comment:
@singalsu One downside is that this memory cannot be freed even if all dmic pipelines are stopped (unless we get dynamic module loading in the future). But this is ~100 bytes, so maybe something that can be optimized later.

Reply:
There will be the NHLT blob that is 4 kB so that at least should be dynamically allocated. I'd like to see the issue #4175 to be fixed by this PR. I'll bring back dynamic allocate in the NHLT PR #4025.